### ISSUE TITLE
Code Editor 2: User-made snippets don't work

### ISSUE THREAD

**Issue by KormexGit** [YoYoGames/GameMaker-Bugs#6344]
### Description
In CE2, user made snippets cannot be accessed in any way, since the F4 menu is gone and they do not appear in autocomplete.

### Steps To Reproduce
Try to use a user defined snippet.

### Which version of GameMaker are you reporting this issue for?
IDE v2024.600.0.576 Runtime v2024.600.0.602

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22631.0

[90b023b1-b594-408d-9fb9-94405408ce50](https://api.gamemaker.io/api/github/downloads/90b023b1-b594-408d-9fb9-94405408ce50)

**Comment by zreedy** [comment]
This feature is now implemented but its implementation might change in the future.
Currently, similar to CE1 behavior, if a snippets.txt file located at `C:\ProgramData\GameMakerStudio2\User\snippets.txt` then it will be loaded and added to the snippets when the Language Server is initialized. Changes to the snippets requires restarting the IDE to see the effects.
The format is that each line is a Snippet entry in the format `NAME:VALUE`, where `NAME` is what's displayed inside of the Code Completion List, and `VALUE` is a [TextMate/LSP 3.17](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#snippet_syntax) compatible syntax. New Lines are escaped as `\n`

Please be mindful that the CE1 snippet syntax used `$` to denote newlines, however this symbol escapes various directives in TextMate Snippet syntax (tab stops, variables, etc.) Any existing user snippets might be misinterpreted in CE2 if they are not converted.

Example
snippets.txt
```
custom:var year = ${CURRENT_YEAR};\nshow_debug_message(year); // ${CURRENT_YEAR}$0
```

Result
![Image](https://github.com/user-attachments/assets/37f0cbc6-08cf-4c5b-8e4e-ce91b92abc7e)
![Image](https://github.com/user-attachments/assets/a41f1c7d-9cb0-433a-bd09-cedc471d3759)


Snippet Syntax Documentation
- https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#snippet_syntax
- https://macromates.com/textmate/manual/snippets

**Comment by zreedy** [comment]
Non-standard Extensions to the Snippet Syntax.

Variables
- `CLIPBOARD`: Contents of the clipboard
- `WORKSPACE_NAME`: Name of current project (yyp filename without extension)
- `WORKSPACE_FOLDER`: Path to yyp
- `CURSOR_INDEX`: Column-1
- `CURSOR_NUMBER`: Column
- `CURRENT_YEAR`: yyyy
- `CURRENT_YEAR_SHORT`: yy
- `CURRENT_MONTH`: MM
- `CURRENT_MONTH_NAME`: M
- `CURRENT_MONTH_NAME_SHORT`: MMM
- `CURRENT_HOUR`: HH
- `CURRENT_MINUTE`: mm
- `CURRENT_SECOND`: ss
- `CURRENT_SECONDS_UNIX`: Now UNIX Timestamp (not UTC)
- `CURRENT_TIMEZONE_OFFSET`: +HHMM
- `RANDOM`: A random base 10 integer between inclusive 100000-999999
- `RANDOM_HEX`: A random base 16 integer between inclusive 0x111111-0xFFFFFF (excluding leading `0x`)
- `UUID`: A new GUIDv4 in lower case with hyphens format

**Comment by zreedy** [comment]
An update for this ticket. We have now implemented support for `*.tmSnippet`s where the snippets are loaded from `C:\ProgramData\GameMakerStudio2\User\*.tmSnippet`. This is the standard format. If there is a CE1 `snippets.txt` in that directory it will be upgraded one time to the `*.tmSnippet` format.

**Comment by sihammill** [comment]
fix verified in Beta IDE v2024.1300.0.727  Runtime v2024.1300.0.739
